src/Protocol/ActivityPub/Transmitter.php

```diff
@@ -0,0 +1,910 @@
+		$conversation = DBA::selectFirst('conversation', ['source'], $condition);
+		if (DBA::isResult($conversation)) {
+			$data = json_decode($conversation['source']);
+			if (!empty($data)) {
+				return $data;
+			}
+		}
+
+		$type = self::getTypeOfItem($item);
+
+		if (!$object_mode) {
+			$data = ['@context' => ActivityPub::CONTEXT];
+
+			if ($item['deleted'] && ($item['gravity'] == GRAVITY_ACTIVITY)) {
+				$type = 'Undo';
+			} elseif ($item['deleted']) {
+				$type = 'Delete';
+			}
+		} else {
+			$data = [];
+		}
+
+		$data['id'] = $item['uri'] . '#' . $type;
+		$data['type'] = $type;
+		$data['actor'] = $item['author-link'];
+
+		$data['published'] = DateTimeFormat::utc($item["created"]."+00:00", DateTimeFormat::ATOM);
+
+		if ($item["created"] != $item["edited"]) {
+			$data['updated'] = DateTimeFormat::utc($item["edited"]."+00:00", DateTimeFormat::ATOM);
+		}
+
+		$data['context'] = self::fetchContextURLForItem($item);
+
+		$data = array_merge($data, self::createPermissionBlockForItem($item));
+
+		if (in_array($data['type'], ['Create', 'Update', 'Announce', 'Delete'])) {
+			$data['object'] = self::createNote($item);
+		} elseif ($data['type'] == 'Undo') {
+			$data['object'] = self::createActivityFromItem($item_id, true);
+		} else {
+			$data['object'] = $item['thr-parent'];
+		}
+
+		$owner = User::getOwnerDataById($item['uid']);
+
+		if (!$object_mode) {
+			return LDSignature::sign($data, $owner);
+		} else {
+			return $data;
+		}
+
+		/// @todo Create "conversation" entry
+	}
+
+	/**
+	 * @brief Creates an object array for a given item id
+	 *
+	 * @param integer $item_id
+	 *
+	 * @return object array
+	 */
+	public static function createObjectFromItemID($item_id)
+	{
+		$item = Item::selectFirst([], ['id' => $item_id, 'parent-network' => Protocol::NATIVE_SUPPORT]);
+
+		if (!DBA::isResult($item)) {
+			return false;
+		}
+
+		$data = ['@context' => ActivityPub::CONTEXT];
+		$data = array_merge($data, self::createNote($item));
+
+		return $data;
+	}
+
+	/**
+	 * @brief Returns a tag array for a given item array
+	 *
+	 * @param array $item
+	 *
+	 * @return array of tags
+	 */
+	private static function createTagList($item)
+	{
+		$tags = [];
+
+		$terms = Term::tagArrayFromItemId($item['id']);
+		foreach ($terms as $term) {
+			if ($term['type'] == TERM_HASHTAG) {
+				$tags[] = ['type' => 'Hashtag', 'href' => $term['url'], 'name' => '#' . $term['term']];
+			} elseif ($term['type'] == TERM_MENTION) {
+				$contact = Contact::getDetailsByURL($term['url']);
+				if (!empty($contact['addr'])) {
+					$mention = '@' . $contact['addr'];
+				} else {
+					$mention = '@' . $term['url'];
+				}
+
+				$tags[] = ['type' => 'Mention', 'href' => $term['url'], 'name' => $mention];
+			}
+		}
+		return $tags;
+	}
+
+	/**
+	 * @brief Adds attachment data to the JSON document
+	 *
+	 * @param array $item Data of the item that is to be posted
+	 * @param text $type Object type
+	 * @return attachment array
```

Comment:
for consistency please use the return format @return type description - `@return array of attachment`